plasmapy/utils/collections_helpers.py

```diff
@@ -0,0 +1,46 @@
+from collections import defaultdict
+
+
+def compare_default_dicts(a: defaultdict, b: defaultdict) -> bool:
+    """Compare two defaultdicts, return True if equal, else False.
+    Does a benign or soft compare. If the defaultdicts COULD become
+    equal, they are considered equal.
+
+    * Does NOT change the memory imprint of any of the dictionaries.
+    * Any overlapping keys, must have same value.
+    * Keys unique to one, must have the default value of the other.
+    * Order of input does NOT matter.
+
+    Example:
+    a = defaultdict(lambda: "", a=42, b=42, c="")
+    b = defaultdict(lambda: 42, c="", d="")
+    compare_defaultdicts(a, b) -> True
```

Comment:
This could be put into docstring form to get tested on each doc build! Take our other functions for examples.